examples/data_types_and_io/data_types_and_io/dataclass.py

```diff
@@ -0,0 +1,135 @@
+
+
+# %% [markdown]
+# ## Python types
+# We define a `dataclass` with `int`, `str` and `dict` as the data types.
+# %%
+@dataclass
+class Datum(DataClassJSONMixin):
+    x: int
+    y: str
+    z: dict[int, str]
+
+
+# %% [markdown]
+# You can send a `dataclass` between different tasks written in various languages, and input it through the Flyte console as raw JSON.
+#
+# :::{note}
+# All variables in a data class should be **annotated with their type**. Failure to do should will result in an error.
+# :::
+#
+# Once declared, a dataclass can be returned as an output or accepted as an input.
+# %%
+@task
+def stringify(s: int) -> Datum:
+    """
+    A dataclass return will be treated as a single complex JSON return.
+    """
+    return Datum(x=s, y=str(s), z={s: str(s)})
+
+
+@task
+def add(x: Datum, y: Datum) -> Datum:
+    """
+    Flytekit automatically converts the provided JSON into a data class.
+    If the structures don't match, it triggers a runtime failure.
+    """
+    x.z.update(y.z)
+    return Datum(x=x.x + y.x, y=x.y + y.y, z=x.z)
+
+
+# %% [markdown]
+# ## Flyte types
+# We also define a data class that accepts {std:ref}`StructuredDataset <structured_dataset>`,
+# {std:ref}`FlyteFile <files>` and {std:ref}`FlyteDirectory <folder>`.
+# %%
+@dataclass
+class FlyteTypes(DataClassJSONMixin):
+    dataframe: StructuredDataset
+    file: FlyteFile
+    directory: FlyteDirectory
+
+
+@task
+def upload_data() -> FlyteTypes:
+    """
+    Flytekit will upload FlyteFile, FlyteDirectory and StructuredDataset to the blob store,
+    such as GCP or S3.
+    """
+    # 1. StructuredDataset
+    df = pd.DataFrame({"Name": ["Tom", "Joseph"], "Age": [20, 22]})
+
+    # 2. FlyteDirectory
+    temp_dir = tempfile.mkdtemp(prefix="flyte-")
+    df.to_parquet(temp_dir + "/df.parquet")
+
+    # 3. FlyteFile
+    file_path = tempfile.NamedTemporaryFile(delete=False)
+    file_path.write(b"Hello, World!")
+
+    fs = FlyteTypes(
+        dataframe=StructuredDataset(dataframe=df),
+        file=FlyteFile(file_path.name),
+        directory=FlyteDirectory(temp_dir),
+    )
+    return fs
+
+
+@task
+def download_data(res: FlyteTypes):
+    assert pd.DataFrame({"Name": ["Tom", "Joseph"], "Age": [20, 22]}).equals(res.dataframe.open(pd.DataFrame).all())
+    f = open(res.file, "r")
+    assert f.read() == "Hello, World!"
+    assert os.listdir(res.directory) == ["df.parquet"]
+
+
+# %% [markdown]
+# A data class supports the usage of data associated with Python types, data classes,
+# flyte file, flyte directory and structured dataset.
+#
+# We define a workflow that calls the tasks created above.
+# %%
+@workflow
+def dataclass_wf(x: int, y: int) -> (Datum, FlyteTypes):
+    o1 = add(x=stringify(s=x), y=stringify(s=y))
+    o2 = upload_data()
+    download_data(res=o2)
+    return o1, o2
+
+
+# %% [markdown]
+# You can run the workflow locally as follows:
+# %%
+if __name__ == "__main__":
+    dataclass_wf(x=10, y=20)
```

Comment:
I would love to have pyflyte run commands next to here and dataclass can be passed as a json file